extensions/pkg/controller/healthcheck/reconciler.go

```diff
@@ -123,6 +123,11 @@ func (r *reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 				return r.resultWithRequeue(), err
 			}
 		}
+		r.logger.Info("Do not perform HealthCheck for extension resource. Shoot is hibernated.", "name", acc.GetName(), "namespace", acc.GetNamespace(), "kind", acc.GetObjectKind().GroupVersionKind().Kind)
+		return reconcile.Result{}, err
+	}
+	if cluster.Shoot.Status.IsHibernated {
```

Comment:
Can't this be combined with line 119?

`if controller.IsHibernated(cluster) || cluster.Shoot.Status.IsHibernated {`

Reply:
That would also update the conditions on the extension again (indicating hibernation). I do not think that this is necessary, as the cluster was previously hibernated, so the conditions are already in the right state. Plus no one would read the updated extensions, as the care-controlller does not read extension conditions when the Shoot is hibernated.

Reply:
I was thinking to keep it in sync/similar to what the care controller does: https://github.com/gardener/gardener/blob/master/pkg/operation/botanist/health_check.go#L905-L908

Reply:
Cannot do the check  `cluster.Shoot.Status.IsHibernated` due to the same reasons as stated [here](https://github.com/gardener/gardener/pull/2249#discussion_r417142588)?